controllers/vrg_volrep.go

```diff
@@ -1847,6 +1847,7 @@ func (v *VRGInstance) restorePVsAndPVCsForVolRep(result *ctrl.Result) (int, erro
 	return count, nil
 }
 
+//nolint:funlen
 func (v *VRGInstance) restorePVsAndPVCsFromS3(result *ctrl.Result) (int, error) {
 	err := errors.New("s3Profiles empty")
```

Comment:
This err will contain the last error calling one of:
- v.reconciler.ObjStoreGetter.ObjectStore()
- v.restorePVsFromObjectStore()
- v.restorePVCsFromObjectStore()

If we fail to restore pvs, pvcs or pvcount/pvccount do not match, we return this err at line 1919.

Since we log all errors during the loop, this change does not fix the issue of logging the same error twice.

I think this function need a rewrite - it is too long and it is too hard to get error handling right with some much code.

I think extracting a helper for restoring from single s3 profile will make it much easier to handle errors.

The helper for restoring from single s3 store can return error early:

func (v *VRGInstance) restorePVsAndPVCsFromS3Profile(profileName string) error {
    get object store...
    if failed:
        return err

    restore pvs...
    if failed:
        return err

    restore pvcs...
    if failed:
        return err
 
    check pvs/pvcs counts...
    if failed:
        return err

    return nil
}

No logs are needed in this function - it just return an error. The function for restoring from all s3 stores can call it twice, logging error for each call.

If we failed to restore from both profiles, we need to return a new error about failing to restore from both profiles, since we already logged the error for every s3 profile.